server/src/embedded-languages/documents-manager.ts

```diff
@@ -0,0 +1,78 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import path from 'path'
+import fs from 'fs'
+
+import { type EmbeddedDocumentInfos, type EmbeddedLanguageType } from './utils'
+
+const EMBEDDED_DOCUMENTS_FOLDER = 'embedded-documents'
```

Comment:
If I'm not mistaken, you currently put the temporary files in the workspace. This is very cumbersome for the user who will get flooded by them, even if they are deleted. I suggest putting your temporary documents in [ExtensionContext.storageUri](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext.storageUri). This will be more manageable for the user to ignore.

Of course, a virtual file will be better in the future.

Reply:
Have a look at this doc: https://code.visualstudio.com/api/extension-capabilities/common-capabilities#data-storage

Reply:
for comparison, I know the C/CPP extension compiles opened files for intellisense features (which can take a lot of space :p). It puts the compiled results in `~/.config/Code/User/workspaceStorage`. This seems like a great practice that would even be satisfying in production if we don't have virtual files yet. Even cleanup is nice to have there

Reply:
you can store them in a subfolder like embeddedLanguages/original/relative/path/original.bb.py. to avoid collisions.
Other option is to hash the original path and use that as a file name if you don't want subfolders.

Reply:
storageUri and hashing sound like two great ideas! I'll experiment to see what I can do with it.

Reply:
done

However I decided to use uuid instead of hashes